src/aics-image-viewer/components/TfEditor/index.tsx

```diff
@@ -143,68 +179,96 @@ const TfEditor: React.FC<TfEditorProps> = (props) => {
-      const point = { x, opacity, color: lastColorRef.current };
+  const dragControlPoint = (draggedIdx: number, x: number, opacity: number): void => {
+    const newControlPoints = [...controlPointsRef.current];
+    const draggedPoint = newControlPoints[draggedIdx];
+    draggedPoint.x = x;
+    draggedPoint.opacity = opacity;
 
-      // add new control point to controlPoints
-      const index = d3.bisector<ControlPoint, ControlPoint>((a, b) => a.x - b.x).left(props.controlPoints, point);
-      setDraggedPointIdx(index);
+    // Remove control points to keep the list sorted by x value
+    const bisector = d3.bisector<ControlPoint, ControlPoint>((a, b) => a.x - b.x);
+    const idxLeft = bisector.left(newControlPoints, draggedPoint);
+    const idxRight = bisector.right(newControlPoints, draggedPoint);
 
-      const newControlPoints = [...props.controlPoints];
-      newControlPoints.splice(index, 0, point);
-      setControlPoints(newControlPoints);
+    if (idxLeft < draggedIdx) {
+      const numPointsToRemove = draggedIdx - idxLeft; // should almost always be 1
+      newControlPoints.splice(idxLeft, numPointsToRemove);
+
+      const newIdx = draggedIdx - numPointsToRemove;
+      setDraggedPointIdx(newIdx);
+      setSelectedPointIdx(newIdx);
+    } else if (idxRight > draggedIdx + 1) {
+      newControlPoints.splice(draggedIdx + 1, idxRight - draggedIdx - 1);
     }
 
-    setSelectedPointIdx(draggedPointIdxRef.current);
+    setControlPoints(newControlPoints);
+  };
+
+  const dragRampSlider = (handle: TfEditorRampSliderHandle, x: number): void => {
+    if (handle === TfEditorRampSliderHandle.Min) {
+      const max = props.ramp[1];
+      setRamp([Math.min(x, max), max]);
+    } else {
+      const min = props.ramp[0];
+      setRamp([min, Math.max(x, min)]);
+    }
+  };
 
-    if (event.button === 0) {
+  const handlePlotPointerDown: React.PointerEventHandler<SVGSVGElement> = (event) => {
+    if (props.useControlPoints) {
+      // Advanced mode - we're either creating a new control point or selecting/dragging an existing one
+      if (draggedPointIdxRef.current === null && event.button === 0) {
+        // this click is not on an existing point - create a new one
+        const [x, opacity] = mouseEventToControlPointValues(event);
+        const point = { x, opacity, color: lastColorRef.current };
+
+        // add new control point to controlPoints
+        const index = d3.bisector<ControlPoint, ControlPoint>((a, b) => a.x - b.x).left(props.controlPoints, point);
+        setDraggedPointIdx(index);
+
+        const newControlPoints = [...props.controlPoints];
+        newControlPoints.splice(index, 0, point);
+        setControlPoints(newControlPoints);
+      } else {
+        // this click is on an existing point - update current points to ref (may have been remapped since last edit)
+        controlPointsRef.current = props.controlPoints;
+      }
+
+      if (typeof draggedPointIdxRef.current !== "string") {
```

Comment:
When would this be a string value?

Reply:
This can either be a number, representing a control point, in "advanced" mode; or either `"min"` or `"max"` representing a slider handle in basic mode. See my response to @toloudis, above:

> ...`props.useControlPoints` would make more semantic sense here. But it also wouldn't cause TypeScript to narrow this type for the following lines...

Since this has now tripped up two people, it might need a comment